homeassistant/components/camera/amcrest.py

```diff
@@ -64,26 +82,84 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
             notification_id=NOTIFICATION_ID)
         return False
 
-    add_devices([AmcrestCam(config, data)])
+    add_devices([AmcrestCam(hass, config, data)])
     return True
 
 
 class AmcrestCam(Camera):
     """An implementation of an Amcrest IP camera."""
 
-    def __init__(self, device_info, data):
+    def __init__(self, hass, device_info, data):
         """Initialize an Amcrest camera."""
         super(AmcrestCam, self).__init__()
+        self._base_url = '%s://%s:%s/cgi-bin' % (
```

Comment:
On the next version 1.0.1 of python-amcrest to be published next week, you can use the builtin method for base the url (https://github.com/tchellomello/python-amcrest/blob/master/src/amcrest/http.py#L49). You can replace lines 95-97 by:
```python
self._data = data
self._base_url = self._data.camera._base_url
```